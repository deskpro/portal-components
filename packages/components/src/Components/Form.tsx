import * as React from 'react';
import { Form as FormikForm } from 'formik';
import classNames from 'classnames';

interface IProps {
  className?: string;
  showHover?: boolean;
  children:   React.ReactNode;
}

const Form = ({
  className = '',
  showHover = true,
  ...props
}: IProps) => (
  <FormikForm
    className={classNames('dp-pc_form', className, { 'show-hover': showHover })}
    {...props}
  />);



export default Form;
