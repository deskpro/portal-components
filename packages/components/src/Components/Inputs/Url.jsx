import Field from '../Field';

class Url extends Field {
  constructor(props) {
    super(props);
    this.type = 'url';
  }
}

export default Url;
