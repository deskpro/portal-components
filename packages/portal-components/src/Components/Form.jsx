import React from 'react';
import { Form as FormikForm } from 'formik';

const Form = ({
  ...props
}) => <FormikForm {...props} />;

export default Form;
