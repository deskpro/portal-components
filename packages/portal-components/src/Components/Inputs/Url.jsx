import Input from './Input';

class Url extends Input {
  constructor(props) {
    super(props);
    this.type = 'url';
  }
}

export default Url;
