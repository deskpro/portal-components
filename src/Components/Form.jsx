import React from 'react';
import { Form as FormikForm } from 'formik';

const Form = ({
  ...props
}) => {
  return <FormikForm {...props} />
};

export default Form;