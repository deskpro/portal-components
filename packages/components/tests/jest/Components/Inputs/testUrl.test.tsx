import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { withFormik } from 'formik';
import Form from '../../../../src/Components/Form';
import Url from '../../../../src/Components/Inputs/Url';
import { describe, expect, it } from '@jest/globals';

interface FormValues {
  url?: string;
}

interface MyFormikProps {
  url?: string;
}

describe('>>> Url --- Snapshot', () => {
  it('+++capturing Snapshot of Url', () => {
    const App = () => (
      <Form>
        <Url name="url" />
      </Form>
    );
    const FormikApp = withFormik<MyFormikProps, FormValues>({
      mapPropsToValues({
        url
      }) {
        return {
          url: url || 'www.deskpro.com'
        };
      },
      handleSubmit(values, { setSubmitting }) {
        setSubmitting(false);
      }
    })(App);
    const renderedValue = renderer.create(<FormikApp />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
