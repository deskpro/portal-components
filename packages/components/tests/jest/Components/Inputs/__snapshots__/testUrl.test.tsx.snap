// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Url --- Snapshot +++capturing Snapshot of Url 1`] = `
<form
  className="dp-pc_form show-hover"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="dp-pc_field"
  >
    <label
      className="dp-pc_label"
      htmlFor="field_1"
    >
      
    </label>
    <input
      className="dp-pc_input"
      id="field_1"
      name="url"
      onBlur={[Function]}
      onChange={[Function]}
      required={false}
      type="url"
      value="www.deskpro.com"
    />
  </div>
</form>
`;
